### PR TITLE
Pr/issue 226 findby type definitions

### PR DESCRIPTION
**Pull request by JaxCavalera**
**What**:
- Addresses #226 including type definitions for findBy and findAllBy

**Why**:
- Helps users of this package working with typescript

**How**:
- Followed the outlined queries documentation to ensure all listed queries now have findBy and findAllBy type definitions.
- Adjusts order so it reads in the same order as listed in the API documentation.

**Checklist**:
- [ ] Documentation added to the [docs site](https://github.com/alexkrolick/testing-library-docs) NA
- [X ] Typescript definitions updated
- [ ] Tests NA
- [X] Ready to be merged

I have left `ByValue`, `BySelectText` as is for now since there is a dedicated issue to removing the deprecated queries.

### CODE REVIEW

File: typings/queries.d.ts [hunk 1]
```diff
@@ -13,12 +13,24 @@ export type AllByBoundAttribute = (
   options?: MatcherOptions,
 ) => HTMLElement[]
 
+export type FindAllByBoundAttribute = (
```
Comment (kentcdodds) [inline typings/queries.d.ts]: The `findAllBy` and `findBy` queries accept 4 arguments. The last one is this: https://github.com/kentcdodds/dom-testing-library/blob/96f21a5219332d7dd058f736afcf29758bbdfc11/typings/wait-for-element.d.ts#L3-L7

Comment (JaxCavalera) [reply typings/queries.d.ts]: No worries I'll make that adjustment now

**Comment by kentcdodds** [comment]
@all-contributors please add @JaxCavalera for code

**Pull request merged**